src/golang/lib/collections/shared/engine.go

```diff
@@ -0,0 +1,40 @@
+package shared
+
+import (
+	"database/sql/driver"
+
+	"github.com/aqueducthq/aqueduct/lib/collections/utils"
+	"github.com/google/uuid"
+)
+
+type EngineType string
+
+const (
+	AqueductEngineType EngineType = "aqueduct"
+	AirflowEngineType  EngineType = "airflow"
+)
+
+type EngineConfig struct {
```

Comment:
TODO: Create task to consider alternative "union" struct approach. (I really don't like this approach, but its the fastest solution for now.)